osu.Desktop/DiscordRichPresence.cs

```diff
@@ -173,6 +189,37 @@ private string truncate(string str)
             return null;
         }
 
+        private void fetchBeatmapSet()
+        {
+            IBeatmapInfo? beatmap = getBeatmap(activity.Value);
+            if (beatmap == null)
+            {
+                beatmapSetOnline.Value = null;
+                return;
+            }
+
+            if (beatmap.BeatmapSet is IBeatmapSetOnlineInfo online)
+            {
+                beatmapSetOnline.Value = online;
+            }
+            else
+            {
+                if (api.IsLoggedIn)
+                {
+                    var req = new GetBeatmapSetRequest(beatmap.OnlineID, BeatmapSetLookupType.BeatmapId);
+                    req.Success += res =>
+                    {
+                        beatmapSetOnline.Value = res;
+                    };
+                    api.Queue(req);
```

Comment:
Should likely be using `BeatmapLookupCache` instead to reduce number of redundant requests for something this low-priority. At which point fetching the image via hardcoded URL pattern may be more desirable? Not sure.

Reply:
This has been changed to use a BeatmapLookupCache instead.